src/Shared/Sdk/Providers/ServiceClients/IClientFactory.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Threading.Tasks;
+using ACMTTU.NoteSharing.Shared.DataContracts;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Storage.Blob;
+
+namespace ACMTTU.NoteSharing.Shared.SDK.Clients
+{
+    public interface IClientFactory
+    {
+        CloudBlobClient GetStorageClient(string connectionString);
```

Comment:
Small suggestion: Do not use string for the connectionString, instead use the ConnectionString class in the stdlib. This makes parsing of the strong unnecessary, it allows you to select individual fields easily, and it avoids having a "string-ly typed" typeclass.

Reply:
I never have to parse it. 

I got this code directly from the [Quickstart](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-quickstart-blobs-dotnet)

It looks like this is the recommended way for connecting to the database/storage accounts

Reply:
I was wrong on what the class was. It is a connectionstring builder: https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/connection-string-builders.

Very strange that there is an EventHubConnectionString, but not one to connect to a CosmosDB: https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-dotnet-standard-getstarted-send

I wonder if the SqlCSB works. Unsure. Either way it's not a big deal.